src/contracts/token/Eigen.sol

```diff
@@ -123,14 +123,14 @@ contract Eigen is OwnableUpgradeable, ERC20VotesUpgradeable {
      */
     function wrap(uint256 amount) external {
         require(bEIGEN.transferFrom(msg.sender, address(this), amount), "Eigen.wrap: bEIGEN transfer failed");
-        _transfer(address(this), msg.sender, amount);
+        _mint(msg.sender, amount);
     }
 
     /**
      * @notice This function allows Eigen holders to unwrap their tokens into bEIGEN
      */
     function unwrap(uint256 amount) external {
-        _transfer(msg.sender, address(this), amount);
+        _burn(msg.sender, amount);
```

Comment:
I think the `_transfer` version of these methods caused us to add `from == this || to == this` to `_beforeTokenTransfer`

Now that we're getting rid of it in favor of `_mint` and `_burn`, that method needs to account for `_beforeTokenTransfer` being called with `from == address(0)` for mint, and `to == address(0)` for burn.

I think only the former of those two is supported - does this mean I won't be able to `unwrap`?

Reply:
you are absolutely correct. fixed in https://github.com/Layr-Labs/eigenlayer-contracts/pull/717/commits/c53aa74b7b2860d30dbf1a9f3f86eca22aa305a2 good catch and thank you for this!